src/sentry/static/sentry/app/views/usageStats/usageStatsOrg.tsx

```diff
@@ -121,41 +135,72 @@ class UsageStatsOrganization extends AsyncComponent<Props, State> {
   } {
     const {dataDatetime} = this.props;
     const {period, start, end} = dataDatetime;
+
     const interval = getInterval(dataDatetime);
+    const intervalMinutes = parsePeriodToHours(interval) * 60;
+    if (intervalMinutes < 1) {
+      throw new Error('UsageStatsOrg: Unable to parse interval for specified period');
+    }
 
-    let chartDateStart = moment().subtract(14, 'd');
-    let chartDateEnd = moment();
+    const FORMAT_DATETIME =
+      intervalMinutes >= 24 * 60 ? FORMAT_DATETIME_DAILY : FORMAT_DATETIME_HOURLY;
 
-    try {
-      if (start && end) {
-        chartDateStart = moment(start);
-        chartDateEnd = moment(end);
-      }
+    if (start && end) {
+      const dateStart = moment(start);
+      const dateEnd = moment(end);
+
+      return {
+        chartDateInterval: interval,
+        chartDateStart: dateStart.format(),
+        chartDateEnd: dateEnd.format(),
+        chartDateStartDisplay: dateStart.local().format(FORMAT_DATETIME),
+        chartDateEndDisplay: dateEnd.local().format(FORMAT_DATETIME),
+      };
+    }
 
-      if (period) {
+    // Default to 14d period at 1hr interval
+    let chartDateStart = moment().subtract(14, 'd');
+    const chartDateEnd = moment();
+
+    if (period) {
+      try {
         const statsPeriod = parseStatsPeriod(period);
         if (!statsPeriod) {
-          throw new Error('Format for data period is not recognized');
+          throw new Error('UsageStatsOrg: Format for data period is not recognized');
+        }
+
+        if (intervalMinutes >= 24 * 60) {
+          chartDateEnd.startOf('d');
+        } else if (intervalMinutes >= 60) {
+          chartDateEnd.startOf('h');
+        } else if (intervalMinutes >= 1) {
+          chartDateEnd.startOf('m');
```

Comment:
Should bucket alignments be done on the server side? In sessions we do it server side via the `get_constrained_date_range` function.

Reply:
Okay, I'll take a look at that in a follow up PR.

Alternatively, the API returns `intervals`, which is an array of the starting datetime for the buckets. I think we can use that to calculate the start/end datetime, which might make this method less bad.

Reply:
Added a ticket into the sprint: https://getsentry.atlassian.net/browse/ER-636

Reply:
Following up here: https://github.com/getsentry/sentry/pull/25435